apis/placement/v1beta1/binding_types.go

```diff
@@ -10,6 +10,49 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	// DispatcherFinalizer is added by the dispatcher to make sure that a binding can only be deleted if the dispatcher
+	// has removed all selected resources from the bound cluster.
+	DispatcherFinalizer = fleetPrefix + "dispatcher-cleanup"
+
+	// SchedulerFinalizer is added by the scheduler to make sure that a binding can only be deleted if the scheduler
+	// has relieved it from scheduling consideration.
+	SchedulerFinalizer = fleetPrefix + "scheduler-cleanup"
+
+	// ActiveBindingLabel is added by the update controller to mark that a binding is active, i.e., the dispatcher
+	// should place resources to it.
+	//
+	// Note that an active binding may not be associated with the latest scheduling policy snapshot or the latest
+	// resource snapshot. It may be up to another controller, e.g., the rolling update controller, to modify the
+	// association (if applicable). In certain cases (e.g., not enough fitting clusters), the binding may not even
+	// has a target cluster.
+	//
+	// Note also that it is not the scheduler's responsibility to add this label, even though it does
+	// reads this label to inform the scheduling cycle..
+	ActiveBindingLabel = fleetPrefix + "is-active-binding"
```

Comment:
The states are kept as labels for easier queries; a field would also work though. Anti-tempering is needed to make sure that the system functions correctly.